xdsl/transforms/varith_transformations.py

```diff
@@ -173,12 +174,8 @@ def is_integer_like_type(t: Attribute) -> bool:
     Returns true if t is an integer/container of integers/container of
     container of integers ...
     """
-    if isinstance(t, builtin.IntegerType | builtin.IndexType):
-        return True
-    if isinstance(t, builtin.ContainerType):
-        elm_type = cast(builtin.ContainerType[Attribute], t).get_element_type()
-        return is_integer_like_type(elm_type)
-    return False
+    t = get_element_type_or_self(t)
+    return isinstance(t, builtin.IntegerType | builtin.IndexType)
```

Comment:
is this the same as `is_integer_like_type`?

Reply:
It *is* `is_integer_like_type` :)